### PR TITLE
Hook up Datavyu-player AVFoundation player

### LINKED ISSUE #122: Remove Native OSX media player
- Move the Native OSX dependencies to the datavyu player repo pom file
- Move the Native OSX player to the datavyu player repo
- Move the native AV Foundation Code to the datavyu player repo

### PR DESCRIPTION
**Pull request by TheoWolf**
closes #122

### CODE REVIEW

File: src/main/java/org/datavyu/plugins/javafx/JavaFxApplication.java [hunk 1]
```diff
@@ -2,177 +2,127 @@
...
+import org.datavyu.plugins.MediaPlayer;
+import org.datavyu.plugins.PlayerStateEvent;
 import org.datavyu.views.VideoController;
 
+import java.awt.*;
 import java.io.File;
+import java.net.URI;
 
-public class JfxApplication extends Application {
+public class JavaFxApplication extends Application {
```
Comment (fraudies) [inline src/main/java/org/datavyu/plugins/javafx/JavaFxApplication.java]: Could we simplify the whole thing by not using javafx?

File: src/main/java/org/datavyu/views/VideoController.java [hunk 2]
```diff
@@ -1606,31 +1606,36 @@ public void jogBackAction() {
...
+            // Update the clock timer with the new time
+            long newTime = clockTime - (clockTime % stepSize) - stepSize;
             for (StreamViewer streamViewer : streamViewers) {
                 // TODO: Tie offset & duration to stream viewer only and pull it in the track model
                 TrackModel trackModel = tracksEditorController.getTrackModel(streamViewer.getIdentifier());
                 // We can only use the step function if this frame rate is close enough to the highest frame rate
                 if (streamViewer.isStepEnabled() && almostEqual(streamViewer.getFramesPerSecond(), frameRate,
                         ALMOST_EQUAL_FRAME_RATES)) {
                     streamViewer.stepBackward();
+                    logger.info("Jog Forward Stream " + streamViewer.getIdentifier() +
```
Comment (fraudies) [inline src/main/java/org/datavyu/views/VideoController.java]: Change log level: info -> debug

Rational: The user can continually press the step button which will clutter the log file

File: src/main/java/org/datavyu/views/VideoController.java [hunk 3]
```diff
@@ -1685,28 +1690,32 @@ public void jogForwardAction() {
...
+            // Update the clock timer with the new time
+            long newTime = clockTime - (clockTime % stepSize) + stepSize;
             for (StreamViewer streamViewer : streamViewers) {
-                // TODO: Tie offset & duration to stream viewer only and pull it in the track model
                 TrackModel trackModel = tracksEditorController.getTrackModel(streamViewer.getIdentifier());
+                // TODO: Tie offset & duration to stream viewer only and pull it in the track model
                 if (streamViewer.isStepEnabled() && almostEqual(streamViewer.getFramesPerSecond(), frameRate,
                         ALMOST_EQUAL_FRAME_RATES)) {
                     streamViewer.stepForward();
+                    logger.info("Jog Forward Stream " + streamViewer.getIdentifier() +
```
Comment (fraudies) [inline src/main/java/org/datavyu/views/VideoController.java]: Change log level: info -> debug

File: src/main/java/org/datavyu/views/VideoController.java (hunk 2 above)
Comment (TheoWolf) [reply src/main/java/org/datavyu/views/VideoController.java]: Done!

File: src/main/java/org/datavyu/views/VideoController.java (hunk 3 above)
Comment (TheoWolf) [reply src/main/java/org/datavyu/views/VideoController.java]: Done!

File: src/main/java/org/datavyu/plugins/javafx/JavaFxApplication.java (hunk 1 above)
Comment (TheoWolf) [reply src/main/java/org/datavyu/plugins/javafx/JavaFxApplication.java]: I will push a separate branch for this matter.

**Pull request merged**